### PR TITLE
Add Java test

### PR DESCRIPTION
**Pull request by a4z**
This add Java test to the hello world example 

Not as pretty as I wish, but a base to start with minimal example in 
Current state:
* use conan to get dependencies (jini)
* handle cmake / djinni generator
* create the required library
* add some tests
* use it from Objective-C, Swift, Java

This minimal example helps me to understand better how to proceed with our testing
And I hope it will also be a help for new user (once I added some docs)

### CODE REVIEW

File: tests/java/dev/xlcpp/djinni/NativeLibLoader.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
+//
```
Comment (a4z) [inline tests/java/dev/xlcpp/djinni/NativeLibLoader.java:1]: this file is actually not used, need to remove it

File: CMakeLists.txt [hunk 2]
```diff
@@ -41,12 +47,58 @@ add_library(storage
...
+
+  PUBLIC_HEADER "${objec_header_generated}"
+
+  RESOURCE "${BUNDLE_RESOURCES}"
+
+#    XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY ${APPLE_ID_DEVELOPMENT_TEAM}
 )
 
+message("---------------------------------")
+message("${objec_header_generated}")
```
Comment (a4z) [inline CMakeLists.txt:80]: remove debug message

**Review by jothepro — COMMENTED** [review]
- nice, I have to check out doctest! I've recently used catch2.
- You are building the same target for use with JNI and ObjC? Thats... cool? 😄

**Comment by a4z** [comment]
yes, that's cool ;-) 
building them together on MacOS allows running all java/objective-c test all at once, on the same platform with 1 build

**Pull request merged**